packages/workspace/src/browser/workspace-service.ts

```diff
@@ -373,6 +373,10 @@ export class WorkspaceService implements FrontendApplicationContribution {
         await this.spliceRoots(this._roots.length, 0, uri);
     }
 
+    async addRoots(...uris: URI[]): Promise<void> {
+        await this.spliceRoots(this._roots.length, 0, ...uris);
+    }
```

Comment:
Should we re-use functionality from `addRoot` rather than copy it's implementation?
Alternatively we can:

```ts
async addRoot(uri: URI | URI[]): Promise<void> {
```

Reply:
I'm happy to go either way. I decided to reimplement it because I could imagine someone having overridden `addRoot` to add additional arguments, and so that the plural would be parallel to `removeRoots`, but it is a bit silly to have both `addRoot` and `addRoots`, so if you vote for just modifying `addRoot` to accept multiple roots, I'm happy with that.

Reply:
@vince-fugnitto I'm also fine with either approach :+1: 
I think it would be fine to allow `addRoot` to accept multiple roots as an argument instead of implementing it as it's own method.

Reply:
> ```ts
> async addRoot(uri: URI | URI[]): Promise<void> {
> ```

@vince-fugnitto, I've implemented this version ^^^ :-)